intel_extension_for_transformers/llm/runtime/graph/models/llama/llama.cpp

```diff
@@ -68,14 +68,29 @@ static bool llama_model_eval_internal(model_context& lctx, const model_token* to
 
   MODEL_ASSERT(!!kv_self.ctx);
 
-  const int n_embd = hparams.n_embd;
+  int n_embd = hparams.n_embd;
   const int n_layer = hparams.n_layer;
   const int n_ctx = hparams.n_ctx;
-  const int n_head = hparams.n_head;
+  int n_head = hparams.n_head;
+  int head_size = n_embd / n_head;
+  int n_head_kv = hparams.n_head_kv;
+
+  bool enable_tp = false;
+#ifdef NE_TP_MODEL
```

Comment:
how about separate a function called ```llama_model_eval_internal_parallel``` to improve readiness, and make it free for optimizations with and without TP?